gensim/sklearn_api/d2vmodel.py

```diff
@@ -72,11 +79,11 @@ def fit(self, X, y=None):
             documents=d2v_sentences, dm_mean=self.dm_mean, dm=self.dm,
             dbow_words=self.dbow_words, dm_concat=self.dm_concat, dm_tag_count=self.dm_tag_count,
             docvecs=self.docvecs, docvecs_mapfile=self.docvecs_mapfile, comment=self.comment,
-            trim_rule=self.trim_rule, size=self.size, alpha=self.alpha, window=self.window,
+            trim_rule=self.trim_rule, size=self.vector_size, alpha=self.alpha, window=self.window,
```

Comment:
All changes of your PR should be in 2 lines of `fit` function, that's all that you needed here.